application/backend/src/main/resources/apps/typerefinery/components/content/ticker/script.js

```diff
@@ -48,43 +36,166 @@ function updateTickerComponent(tickerData, id) {
+  );
+
+  // listen when the client is ready
+  window.addEventListener(
+    window.MessageService.Client.events.READY,
+    function (message) {
+      console.log(
+        `ui event: ${window.MessageService.Client.events.READY}`,
+        message.detail
+      );
     }
-
-    // Rendering the template
-    getDataFromDataSource(defaultData, dataSourcePath, id, dataSourceReferenceTime);
-
+  );
+
+  // listen for client id update
+  window.addEventListener(
+    window.MessageService.Client.events.CLIENT_ID,
+    function (message) {
+      console.log(
+        `ui event: ${window.MessageService.Client.events.CLIENT_ID}`,
+        message.detail
+      );
+      document.querySelector("#ws-id").textContent = message.detail;
+    }
+  );
+
+  // listen for messages
+  window.addEventListener(
+    window.MessageService.Client.events.MESSAGE,
+    function (message) {
+      console.log(
+        `ui event: ${window.MessageService.Client.events.MESSAGE}`,
+        message.detail,
+        "hello"
+      );
+      var messageText = message.detail;
+      if (typeof message.detail === "object") {
+        messageText = JSON.stringify(message.detail);
+        var messageType = message.detail.type;
+
+        if (messageType === "meta") {
+          var subscribers = message.detail.subscribers;
+          var publishers = message.detail.publish;
+          var calls = message.detail.call;
+
+          // for each object in subscribers add object id to subscribers select element
+          for (var subscriber in subscribers) {
+            let option = document.createElement("option");
+            option.value = subscribers[subscriber];
+            option.text = subscribers[subscriber];
+            document.getElementsByName("subscribers")[0]?.add(option);
+          }
+
+          // for each object in publishers add object id to publishers select element
+          for (var publisher in publishers) {
+            let option = document.createElement("option");
+            option.value = publishers[publisher].id;
+            option.text = publishers[publisher].id;
+            document.getElementsByName("publishers")[0]?.add(option);
+          }
+
+          // for each object in calls add object id to calls select element
+          for (var call in calls) {
+            let option = document.createElement("option");
+            option.value = calls[call].id;
+            option.text = calls[call].id;
+            document.getElementsByName("calls")[0]?.add(option);
+          }
+        }
+      }
+      var messageItem = document.createElement("li");
+      var content = document.createTextNode(messageText);
+      messageItem?.appendChild(content);
+      const payload = message.detail.data.payload;
+      const parsedPayload = JSON.parse(payload);
+      const resultPayload = parsedPayload.data;
+      document.getElementById("messages")?.appendChild(messageItem);
+      updateTickerComponent(resultPayload, component);
```

Comment:
I think we can use the DOM to get the component because in the future these functions will be made common for all the components, so we can use the DOM to fetch the component

Reply:
Done